src/irisnet/CMakeLists.txt

```diff
@@ -58,88 +50,47 @@ set(SOURCES
 )
 
 if(UNIX)
-    list(APPEND SOURCES corelib/netinterface_unix.cpp)
+    target_sources(irisnet PRIVATE
+        corelib/netinterface_unix.cpp
+    )
 endif()
 
-set(HEADERS
-    corelib/irisnetexport.h
```

Comment:
Will it make all these headers invisible in an QtCreator in project mode?
I guess it may also break moc (@Vitozz wdyt?)

Reply:
I don't know about QtCreator. MOC definitely isn't broken, it works automagically with CMake.

Reply:
I just tried to link with iris with all these changes and it seems moc didn't work. Some symbols (e.g signals) are missing.

Reply:
and yes after re-adding headers to cmake files the symbols are back.